packages/p2p/src/hooks/useP2PExchangeRate.ts

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useFetch } from '@deriv/api';
+import { useExchangeRate } from '@deriv/hooks';
+
+const useP2PExchangeRate = () => {
+    const { getRate } = useExchangeRate();
+    const { data, ...rest } = useFetch('website_status');
+    const override_exchange_rate = data?.website_status?.p2p_config?.override_exchange_rate;
+
+    const getP2PRate = React.useCallback(
+        (currency: string) => (override_exchange_rate ? Number(override_exchange_rate) : getRate(currency)),
+        [getRate, override_exchange_rate]
+    );
+
+    return { getP2PRate, ...rest };
```

Comment:
```suggestion
    const getP2PRate = React.useCallback(
        (currency: string) => (override_exchange_rate ? Number(override_exchange_rate) : getRate(currency)),
        [getRate, override_exchange_rate]
    );
```

You can change calculate it here 🤔  that way you just return the `p2p_rate` instead of calling `getP2PRate()` everywhere you want to use it.